examples/systemcalls/systemcalls.c

```diff
@@ -92,7 +140,57 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
  *   The rest of the behaviour is same as do_exec()
  *
 */
+    // Open the outputfile with right permissiom
```

Comment:
@mubeena12 think about how you could avoid duplicating code between these two implementations